puma/histogram.py

```diff
@@ -34,12 +36,21 @@ def __init__(
         Parameters
         ----------
         values : np.ndarray
-            Input data for the histogram
+            Input data for the histogram. If bin_edges is specified (not None)
+            then this array is treated as the bin heights.
         weights : np.ndarray, optional
             Weights for the input data. Has to be an array of same length as the input
             data with a weight for each entry. If not specified, weight 1 will be given
             to each entry. The uncertainties are calculated as the square root of the
             squared weights (for each bin separately). By default None.
+        bin_edges : np.ndarray, optional
+            If specified, the histogram is considered "filled": the array given to 
+            values is treated as if it was the bin heights corresponding to these
+            bin_edges and the "weights" input is ignored. By default None. 
+        sum_squared_weights : np.ndarray, optional
```

Comment:
I think you can just keep the weights keyword, as if you specify `bin_edges` we already know we are using the other type of histogram

Reply:
Sorry I am not sure what you mean by keeping the weights keyword. You mean using it instead of `sum_squared_weights` implicitely like we do for `values` as bin_heights?

Reply:
Sorry for being unclear but yes, exactly this

Reply:
I understand that it would make the logic cleaner with less parameters but couldn't it get a bit confusing to pass the sum of squared weights of each bin to a parameter called `weights`? If you are sure I can change it quickly but I want to make sure that it doesn't make things confusing

Reply:
I don't mind having a separate argument, but we should then take advantage of this and add a couple more checks:

1. that `weights` and `sum_squared_weights` are not set at the same time
2. that `weights` is unset if `bin_edges`is set

Reply:
@zcapjdb are you happy if we use the `sum_squared_weights` as suggested by @Choklatier? Assuming the above checks are added

Reply:
Yeah of course, those checks sound reasonable and prevent any ambiguities